MSAL/test/integration/native_auth/end_to_end/MSALNativeAuthEndToEndBaseTestCase.swift

```diff
@@ -71,14 +77,51 @@ class MSALNativeAuthEndToEndBaseTestCase: XCTestCase {
     
     func initialisePublicClientApplication(
         clientIdType: ClientIdType = .password,
-        challengeTypes: MSALNativeAuthChallengeTypes = [.OOB, .password]
+        challengeTypes: MSALNativeAuthChallengeTypes = [.OOB, .password],
+        customSubdomainFormat: Int? = nil
```

Comment:
Using an Int in this case create the following issues:
- Int, if not validated, can led to "arrray index out of bound error"
- The caller of the method will need to check all the formats available before passing the correct index
- Changing the order of the format in the array can led to use the unexpected custom domain.

Did you take in consideration using an enum instead?
I suggest using an enum like this one:
```
enum AuthorityURLFormat {
    case tenantSubdomainShortVersion
    case tenantSubdomainLongVersion
    case tenantSubdomainTenantId
}
```

Reply:
Good idea. Done.